components/calendar/src/astronomy.rs

```diff
@@ -0,0 +1,657 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// This file contains important structs and functions relating to location,
+/// time, and astronomy; these are intended for calender calculations and based off
+/// _Calendrical Calculations_ by Reingold & Dershowitz.
+use crate::iso::Iso;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::{Date, Gregorian};
+
+#[derive(Debug)]
+pub struct Location {
+    latitude: f32,  // latitude from -90 to 90
+    longitude: f32, // longitude from -180 to 180
+    elevation: f32, // elevation in meters
+}
+
+// The mean synodic month in days of 86400 atomic seconds
+// (86400 = 24 * 60 * 60)
+pub const MEAN_SYNODIC_MONTH: f64 = 29.530588861;
+
+// The Moment of noon on January 1, 2000
+pub const J2000: Moment = Moment::new(730120.5);
+
+impl Location {
+    /// Create a location; latitude is from -90 to 90, and longitude is from -180 to 180
+    ///
+    /// ```rust
+    /// use icu::calendar::astronomy::Location;
+    ///
+    /// let location: Location = Location::new(29.3, -132.6, 1032.5);
+    /// ```
+    pub fn new(latitude_input: f32, longitude_input: f32, elevation_input: f32) -> Location {
```

Comment:
suggestion: maybe this could be a `try_new` returning a `Result` given that 1000 degrees latitude and 500 degrees longitude doesn't make sense.

Another alternative would be to use modulo arithmetic to convert longitude input values into an equivalent value between -180 and 180. But would that makes sense for latitude? If not, then you still would need a Result for that reason, rather than a plain Location.